database.go

```diff
@@ -97,11 +105,37 @@ type databaseFormat struct {
 	Time  time.Time
 }
 
+// ErrDBNotReady is a public error type with a wait method for database.
+type ErrDBNotReady struct {
+	cause error
+	wg    chan int
+}
+
+func (e *ErrDBNotReady) Error() string {
+	return fmt.Sprintf("Database not ready. (%v)", e.cause)
+}
+
+// Wait blocks and waits until the database is ready or until it times out.
+// If the timeout value is 0, it waits indefinitely.
+func (e *ErrDBNotReady) Wait(timeout time.Duration) bool {
```

Comment:
This should really be a `context.Context`. It provides significantly more flexibility than just a `time.Duration`